crates/modelardb_storage/src/parser.rs

```diff
@@ -47,11 +49,117 @@ use sqlparser::tokenizer::Token;
+        match statement {
+            Statement::CreateTable(create_table) => semantic_checks_for_create_table(create_table),
+            Statement::Drop {
+                object_type,
+                if_exists,
+                names,
+                cascade,
+                restrict,
+                purge,
+                temporary,
+            } => {
+                let table_names = semantic_checks_for_drop(
+                    object_type,
+                    if_exists,
+                    names,
+                    cascade,
+                    restrict,
+                    purge,
+                    temporary,
+                )?;
+                Ok(ModelarDbStatement::DropTable(table_names))
+            }
+            Statement::Truncate {
+                table_names,
+                partitions,
+                table,
+                only,
+                identity,
+                cascade,
+            } => {
+                let table_names = semantic_checks_for_truncate(
+                    table_names,
+                    partitions,
+                    table,
+                    only,
+                    identity,
+                    cascade,
+                )?;
+                Ok(ModelarDbStatement::TruncateTable(table_names))
+            }
+            Statement::Explain { .. } => Ok(ModelarDbStatement::Statement(statement)),
+            Statement::Query(ref boxed_query) => {
+                if let Some(addresses) = extract_include_addresses(boxed_query) {
+                    Ok(ModelarDbStatement::IncludeSelect(statement, addresses))
+                } else {
+                    Ok(ModelarDbStatement::Statement(statement))
+                }
+            }
+            Statement::Insert(ref _insert) => Ok(ModelarDbStatement::Statement(statement)),
+            _ => Err(ModelarDbStorageError::InvalidArgument(
+                "Only CREATE, DROP, TRUNCATE, EXPLAIN, SELECT, and INSERT are supported".to_owned(),
+            )),
+        }
+    }
+}
+
+/// Parse `sql_expression` into a [`DFExpr`] if it is a correct SQL expression that only references
+/// columns in [`DFSchema`], otherwise [`ModelarDbStorageError`] is returned.
+pub fn tokenize_and_parse_sql_expression(
+    sql_expression: &str,
+    df_schema: &DFSchema,
+) -> Result<DFExpr> {
+    let context_provider = ParserContextProvider::new();
+    let sql_to_rel = SqlToRel::new(&context_provider);
+    let mut planner_context = PlannerContext::new();
+
+    let dialect = ModelarDbDialect::new();
+    let mut parser = Parser::new(&dialect).try_with_sql(sql_expression)?;
+    let parsed_sql_expr = parser.parse_expr()?;
+    sql_to_rel
+        .sql_to_expr(parsed_sql_expr, df_schema, &mut planner_context)
+        .map_err(|error| error.into())
+}
 
 /// SQL dialect that extends `sqlparsers's` [`GenericDialect`] with support for parsing CREATE MODEL
-/// TABLE table_name DDL commands.
+/// TABLE table_name DDL statements.
```

Comment:
Maybe also mention that it extends it to support `INCLUDE` as well.

Reply:
Fixed.